packages/units/src/useUnits.ts

```diff
@@ -23,10 +23,12 @@ import { identity, useContextSelector } from '@googleforcreators/react';
  * Internal dependencies
  */
 import Context from './context';
+import type { State } from './types';
 
-// TODO: Fix selector type.
-function useUnits(selector: () => unknown) {
-  return useContextSelector(Context, selector ?? identity);
+function useUnits(): State;
+function useUnits<T>(selector: (state: State) => T): T;
+function useUnits<T>(selector: (state: State) => T | State = identity) {
```

Comment:
This looks strange.

Reply:
It‘s called function overloading. Makes the types work nicely with context selectors. We use it in the rich-text package too.